### PR TITLE
Species-blocking system for specific jobs

### PR DESCRIPTION
**Pull request by noverd**
## About the PR

In this PR, I added a species-blocking system for specific jobs. This can be useful for blocking aliens from the positions of captain or head. To do this, a new field called "allowSpecies" has been added to the Job prototype. It takes a list of allowed strings. If a player tries to enter a task with a prohibited species, the system will try to find a character in the preferences with a suitable Species. If it doesn't find one, it will create a new character with the allowed species.

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Species-blocking system for specific jobs

### CODE REVIEW

**Comment by noverd** [comment]
> > This can be useful for blocking aliens from the positions of captain or head
> 
> jester-that-helps would like to have a word with you
My discord is CodTenAlt#9052

**Comment by GoodWheatley** [comment]
Why are you trying to codify speciesism

**Comment by mirrorcult** [comment]
cause some servers will want this functionality?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by noverd** [comment]
> Why are you trying to codify speciesism

No discrimination, I just don't think there will be HoS dion in reality

**Comment by GoodWheatley** [comment]
> > Why are you trying to codify speciesism
> 
> No discrimination, I just don't think there will be HoS dion in reality

Didn’t notice that this is just adding the functionality, sorry. Good PR

File: Content.Server/GameTicking/GameTicker.Spawning.cs [hunk 1]
```diff
@@ -165,6 +167,30 @@ private void SpawnPlayer(IPlayerSession player, HumanoidCharacterProfile charact
...
+                            characterNotFound = false;
+                            break;
+                        }
+                    }
+                    if (characterNotFound){
+                        _chatManager.DispatchServerMessage(player, Loc.GetString("game-ticker-player-cant-find-suitable-char-in-preferences"));
+                        character = HumanoidCharacterProfile.RandomWithSpecies();
+                    }
+                }
+            }
```
[the remark this answers is not in the document]
Comment (noverd) [reply Content.Server/GameTicking/GameTicker.Spawning.cs:191]: OK

**Comment by Myakot** [comment]
Finally. Space-station-14 will reach it's perfect state - xenophobia.

**Comment by noverd** [comment]
In the future, when I do this, I will do a PR

**Pull request closed without merging**